### ISSUE TITLE
Custom definitions support

### ISSUE THREAD

**Issue by harrytallbelt** [harrytallbelt/verifiable-tests#30]
Can we allow the user (teacher) to add their own definitions to simplify description of the task? For example
```
{
  "precondition": "n = N && x = X && equalArrays(a, A) && contains(A, X, 0, N-1)",
  ...
  "definitions": [
  {
    "name": "contains",
    "arguments": [ "a", "x", "from", "to" ],
    "definition": "(E k : from <= k <= to : a[k] = x)"
  },
  ...
  ]
  ...
}
```
Would this be consistent with `perm` - a predicate that we do not define, only describe some of its properties?

**Comment by harrytallbelt** [comment]
When implemented, actually use the feature to shorten the specs of the tasks we already have.

**Comment by harrytallbelt** [comment]
Implemented in PR #44
